src/backend/access/transam/xlogarchive.c

```diff
@@ -191,6 +192,14 @@ RestoreArchivedFile(char *path, const char *xlogfname,
 					StrNCpy(dp, contentid, endp - dp);
 					dp += strlen(dp);
 					break;
+				case 'd':
```

Comment:
The same question like about the pgarch_archiveXlog function. 

Construction of the command to be executed in pgarch_archiveXlog() very similar to construction in  RestoreArchivedFile(). Can we create one function to call from pgarch_archiveXlog() and RestoreArchivedFile()?

Reply:
Smells like an over-optimization here. Will hold this comments until future suggestions.

Reply:
This part of code was heavily refactored in the upstream Postgres, but still have separated switch-case blocks. I think, it would be better to address such changes to upstream Postgres first.